File: CvManagementSystem/CvManagementSystem.Api/Contracts/Attributes/CreateUpdateAttributeValueRequest.cs
using UserService.Domain.Models.Attributes;

namespace UserService.Api.Contracts.Attributes;

public class CreateUpdateAttributeValueRequest
{
    public Guid AttributeDefinitionId { get; set; }
    public string? StringValue { get; set; } = null;
    public string? MarkDownValue { get; set; } = null;
    public IFormFile? ImageValue { get; set; } = null;
    public decimal? NumericValue { get; set; } = null;
    public DateOnly? DateValue { get; set; } = null;
    public DateOnly? PeriodStartValue { get; set; } = null;
    public DateOnly? PeriodEndValue { get; set; } = null;
    public bool? BooleanValue { get; set; } = null;
    public Guid? OneOfManyValueId { get; set; }
}

File: Frontend/src/models/CreateUpdateAttributeValueRequest.ts
export interface CreateUpdateAttributeValueRequest {
    attributeDefinitionId: string;
    stringValue?: string | null;
    markDownValue?: string | null;
    imageValue?: File | null;
    numericValue?: number | null;
    dateValue?: string | null;
    periodStartValue?: string | null;
    periodEndValue?: string | null;
    booleanValue?: boolean | null;
    oneOfManyValueId?: string | null;
}